src/Spago/Watch.hs

```diff
@@ -191,3 +201,13 @@ globToParent glob = go pathHead pathRest
     go acc ("**":_rest) = acc
     go acc [_file]      = acc
     go acc (h:rest)     = go (acc </> h) rest
+
+
+-- | Check if the path is ignored by git
+--
+-- `git check-ignore` exits with 1 when path is not ignored, and 128 when
+-- a fatal error occurs (i.e. when not in a git repository).
+isIgnored :: MonadIO m => Text -> m Bool
```

Comment:
Let's add a `FIXME` about unifying this with the other implementation that we have in `Spago.Git`, together with the reason why there is another implementation

Reply:
What would be a good way to unify these though? To be honest I pretty like the idea of `HasGit` constraint. However as you mentioned before, I can't think of a way to make it work with a CLI flag right now. What could be a potential way to unify these, without ditching the one in `Spago.Git`? 🤔

Reply:
Let's move this one to the `Spago.Git` module, rename it `unsafeIsIgnored`, have `Spago.Git.isIgnored` just be an alias to it, and add a `FIXME` comment to remember to remove the unsafe variant. I have some ideas how to make things tidy (e.g. port the codebase to `generic-lens` so that we can just pass in constraints dynamically without having to use custom typeclasses), but it's kind of a big refactoring so this is probably the quickest way to be done with this.